### DISCUSSION TITLE
Master Client ID

### DISCUSSION THREAD

**Discussion by peebeejay** [gnzsnz/ib-gateway-docker#74] [Q&A] [answered]
Appreciate all the work that's gone into this package. 

I'm wondering if there's a nice out of the box way to set the master client ID that isn't specified in the docs? I know that it needs to be defined in the IBC config, but seems like it isn't set at the moment.

**Comment by gnzsnz** [comment]
Hi,

you mean IBC's OverrideTwsMasterClientID option, right?

https://interactivebrokers.github.io/tws-api/initial_setup.html#master_client

I have not included all IBC options, as most of them are really useful in very specific cases.

you can:

- define TWS_SETTINGS_PATH, and a volume for your settings. then you change that setting through ibgateway/tws gui
- you can use CUSTOM_CONFIG=yes, that will require that you mount a config.ini and jts.ini file
- create a PR with the changes, this [line](https://github.com/gnzsnz/ib-gateway-docker/blob/b6d004b2dc10e690c75d5f30b057274d8e39595f/image-files/scripts/common.sh#L9C3-L9C11) is handling env substitution. it should work out of the box, we just need an updated config.ini.tmpl, docker-compose sample file, and README

**Comment by peebeejay** [comment, marked as answer]
Ah yes correct, thank you for the helpful answer. I will see what I can do
